utils/changelog/filter.js

```diff
@@ -0,0 +1,41 @@
+function partition(arr, testItem) {
+  let trueResults = [];
+  let falseResults = [];
+
+  arr.forEach(item => {
+    let result = !!testItem(item);
+    if (result) {
+      trueResults.push(item);
+    } else {
+      falseResults.push(item);
+    }
+  });
+
+  return [trueResults, falseResults];
+}
+
+/**
+ * Filters out "Revert" commits as well as the commits they revert, if found.
+ */
+function filterReverts(commits) {
+  const [revertCommits, filteredLogCommits] = partition(
+    commits,
+    commit => commit.revertsHashes.length
+  );
+
+  let hashesToRemove = [];
+  revertCommits.forEach(revertCommit => {
+    hashesToRemove = hashesToRemove.concat(revertCommit.revertsHashes);
+  });
+
+  return filteredLogCommits.filter(
+    commit => !hashesToRemove.includes(commit.hash)
+  );
```

Comment:
I need to update this for the case where reverted commits were not found in the current changelog.